### PR TITLE
Retrieve separate train/val/test datasets

### PR DESCRIPTION
**Pull request by tatianawu**
Updated `load_dataset` to retrieve a specific dataset split: either train, val, or test. The training script now includes a validation set.

### CODE REVIEW

File: usl_models/usl_models/flood_ml/dataset.py [hunk 1]
```diff
@@ -112,16 +115,17 @@ def load_dataset_windowed(
 ) -> tf.data.Dataset:
     """Creates a dataset which generates chunks for flood model training.
 
-    This dataset produces the input for `model.call`.
-    For getting data to input into `model.call_n`, use `load_dataset` instead.
+    This dataset produces the input for `model.call` and should only be used
+    for training, since it uses labels.
+    For getting dat to input into `model.call_n`, use `load_dataset` instead.
```
Comment (waltaskew) [inline usl_models/usl_models/flood_ml/dataset.py:120]: typo, dat -> data

**Review by waltaskew — APPROVED** [review]

File: usl_models/usl_models/flood_ml/dataset.py (hunk 1 above)
Comment (tatianawu) [reply usl_models/usl_models/flood_ml/dataset.py]: Oops, fixed :)
